dashboard/app/models/pd/fit_weekend_registration_base.rb

```diff
@@ -16,5 +16,139 @@
 #
 
 class Pd::FitWeekendRegistrationBase < ActiveRecord::Base
+  include Pd::Form
+
   self.table_name = 'pd_fit_weekend_registrations'
+
+  belongs_to :pd_application, class_name: 'Pd::Application::ApplicationBase'
+
+  after_initialize :set_registration_year
+  before_validation :set_registration_year
+
+  after_create :update_application_status
+  def update_application_status
+    pd_application.update!(status: 'withdrawn') unless accepted?
+  end
+
+  after_create :send_fit_weekend_confirmation_email
+  def send_fit_weekend_confirmation_email
+    Pd::FitWeekend1819RegistrationMailer.confirmation(self).deliver_now
+  end
+
+  # Override in derived classes and set to valid value.
+  # Setting to nil here fails those validations and prevents this base class from being saved.
+  def set_registration_year
+    self.registration_year = nil
+  end
+
+  YES = 'Yes'.freeze
+  NO = 'No'.freeze
+  YES_OR_NO = [YES, NO].freeze
+
+  def self.options
+    {
+      ableToAttend: YES_OR_NO,
+      dietaryNeeds: [
+        'None',
+        'Vegetarian',
+        'Vegan',
+        'Kosher',
+        'Halal',
+        'Gluten Free',
+        'Food Allergy',
+      ],
+      liveFarAway: YES_OR_NO,
+      addressState: get_all_states_with_dc.to_h.values,
+      howTraveling: [
+        'I will drive by myself',
+        'I will carpool with another FiT Weekend attendee (Please note who)',
+        'Flying',
+        'Amtrak or regional train service',
+        'Public transit (e.g., city bus or light rail)',
+      ],
+      needHotel: YES_OR_NO,
+      needAda: YES_OR_NO,
+      photoRelease: [YES],
+      liabilityWaiver: [YES],
+    }.freeze
+  end
```

Comment:
A lot of the content in here seems potentially year-specific; what's the long-term goal here?

Reply:
I think much of what is in here will be relevant in future years, so I am erring on the side of starting with it in the base class and in future years we can move pieces that aren't being shared into year-specific classes, rather than pulling shared pieces back up into the parent from the year-specific classes.